src/components/VisibilityObserver/VisibilityObserver.tsx

```diff
@@ -1,36 +1,30 @@
 /*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
 import '../../yamui';
 import * as React from 'react';
-import Observer from 'react-intersection-observer';
+import Observer from '@researchgate/react-intersection-observer';
 import { BaseComponentProps } from '../../util/BaseComponent/props';
 
 export interface VisibilityObserverProps extends BaseComponentProps {
   /**
    * A callback which will be triggered when the component is scrolled into view.
    */
-  onEnterView?: (() => void);
+  onEnterView?: ((entry?: IntersectionObserverEntry) => void);
 
   /**
    * A callback which will be triggered when the component is scrolled out of view.
    */
-  onLeaveView?: (() => void);
+  onLeaveView?: ((entry?: IntersectionObserverEntry) => void);
 
   /**
    * Render prop to return child content when the component is visible in the viewport. Once the component
    * has been in view it will always use this render prop, even when scrolled back out of view.
    */
-  renderInView?: (() => React.ReactNode);
+  renderInView?: (() => React.ReactElement<HTMLElement>);
```

Comment:
That sucks, they should be using `ReactNode`.

Reply:
It looks like it's because they use `React.Children.only`, I suspect to enforce a single child.
https://github.com/researchgate/react-intersection-observer/blob/78d5e1d989f6428172eb967904ea81e28472c492/src/IntersectionObserver.js#L167

Reply:
Ah, fair enough.